src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -56,40 +55,52 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
             {
                 throw;
             }
-            catch
+            catch (Exception e)
             {
-                string message = String.Format(CultureInfo.CurrentCulture,
-                    "Invalid storage account '{0}'. Please make sure your credentials are correct.",
-                    account.Credentials.AccountName);
-                throw new InvalidOperationException(message);
-            }
-
-            if (isPrimaryAccount)
-            {
-                // Primary storage accounts require Queues
-                IStorageQueueClient queueClient = account.CreateQueueClient();
-                IStorageQueue queue = queueClient.GetQueueReference("name");
-                try
+                var storageException = e as StorageException;
+                if (storageException != null && storageException.RequestInformation != null && 
+                    storageException.RequestInformation.HttpStatusCode == 400 &&
+                    storageException.RequestInformation.ExtendedErrorInformation.ErrorCode == "InvalidQueryParameterValue")
                 {
-                    await queue.ExistsAsync(cancellationToken);
-                    _primaryCredentials = account.Credentials;
+                    // WARNING: HACK INCOMING
```

Comment:
Can you change this warning to a TODO pointing to a new bug number, tracking getting the storage team to give us a better way of identifying this? I believe you mentioned there was an issue in their repo related to this.